Grid.js

```diff
@@ -0,0 +1,168 @@
+import Cell from "./Cell.js";
+
+class Grid {
+    constructor(
+        gridSize,
+        numMines,
+    ) 
+    {
+        this.gridSize = gridSize;
+        this.numMines = numMines;
+        this.cells = [];
+    }
+
+    createGridCells() {
+        this.cells = new Array(this.gridSize);
+    
+        for (let i = 0; i < this.gridSize; i++) {
+            this.cells[i] = new Array(this.gridSize);
+        };
+    
+        for (let row = 0; row < this.gridSize; row++) {
+            for (let col = 0; col < this.gridSize; col++) {
+                let cell = new Cell(
+                    row,
+                    col,
+                    false,
+                    undefined,
+                    false,
+                    false
+                );
+    
+                this.cells[row][col] = cell;
+            };
+            
+        };
+    };
+
+    getCell(row, col) {
+        return this.cells[row][col];
+    }
+
+    changeCellProperty(row, col, prop, value){
+        this.cells[row][col][prop] = value;
+    }
+    
+    assignMines(){  
+        function getRandomInt(max) {
+            return Math.floor(Math.random() * max);
+        }
+        for (let i = 0; i < this.numMines;) {
+            let row = getRandomInt(this.gridSize);
+            let col = getRandomInt(this.gridSize);
+            let cell = this.getCell(row, col);
+            if (!cell.isMine) {
+                cell.isMine = true;
+                i++;
+            }
+        }
+    };
+
+    applyFnToNeighbors(row, col, callback){
+        let topLeft = {row: row-1, col: col-1};
+        let topCenter = {row: row-1, col: col};
+        let topRight = {row: row-1, col: col+1};
+        
+        let centerLeft = {row: row, col: col-1};
+        let centerRight = {row: row, col: col+1};
+    
+        let bottomLeft = {row: row+1, col: col-1};
+        let bottomCenter = {row: row+1, col: col};
+        let bottomRight = {row: row+1, col: col+1};
+    
+        let surroundingCells = [topLeft, topCenter, topRight, 
+                                centerLeft, centerRight, 
+                                bottomLeft, bottomCenter, bottomRight];
+        //console.log(surroundingCells);
+        
+        surroundingCells.forEach(c => callback(c));
+    };
+
+    isInBounds(coordinate){
+        return coordinate >= 0 && coordinate < this.gridSize;
+    };
+
+    checkForMine(surroundingCell) {
+        if (!this.isInBounds(surroundingCell.row) || !this.isInBounds(surroundingCell.col)) {
+            return 0;  
+        }
+        return this.getCell(surroundingCell.row, surroundingCell.col).isMine ? 1 : 0;
+    };
+
+    getNumMines(cell, row, col){
+        this.getCell(row,col).numMinesAround += this.checkForMine(cell);
+    };
```

Comment:
I think the intent could be clearer here, since the function only returns `0` or `1` I think a boolean is more appropriate, and the naming could be improved. the `isMine` function doesn't need to know that it's being passed a `surroundingCell`, all it needs to know is that it's getting a cell.

also, functions prefixed with `get` usually don't update values, they just _read_ values, so this should be named accordingly.
```suggestion
    isMineWithBoundsCheck(cell) {
        if (!this.isInBounds(cell.row) || !this.isInBounds(cell.col)) {
            return false;  
        }
        return this.getCell(cell.row, cell.col).isMine;
    };

    updateMineCount(cell, row, col) {
        if (this.isMineWithBoundsCheck(cell)) {
          this.getCell(row, col).numMinesAround++;
        }
    };
```

Reply:
cleaner coding style - talked about